### DISCUSSION TITLE
[Environments] - Empty env variables after creation of a new environment

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#61] [Ideas]
I'm on fire today 🔥 
After the creation of a new environment the user must manually add all the environment variables.
In projects with a lot of them this can be a very boring and error prone activity.

I think we can improve this, following some proposal:
* clone the values of another environment of the same level &rarr; I create a new `non-production` environment, all the variables are copied from another casual environment. Better, can we ask to the user from which environment copy the variables
* add this feature inside the `environment variables` section directly, like a button to copy all values from another environment (all values, empty values only, we will see)

I am talking about the `Public variables` mainly, because are the ones integrated inside the Mia-Platform console; I don't know if we can (and we want) to extend this to the `Secreted variables` too.

**Comment by davidebianchi** [comment]
Hi @silversoul93. For the secreted variables, there could be some problems due to who can see the values. We should think about it, sure!
For the Public variables, we have in roadmap to improve the creation of environments to also integrate the runtime setup to avoid all the manual activities. And this could be an interesting idea. But the public variables are per branch, so we should set for all branches of the project.
So, the first suggestion is quite complicated to address. But the second I suppose it would be very useful, we should implement the clone feature!

**Reply by frengooo** [reply]
The clone feature would be very nice 😄 must I create the task inside the roadmap?

**Reply by Danielecina** [reply]
Nice idea!

**Comment by akelity** [comment]
This feature has been positively evaluated and may be considered for inclusion in the roadmap in the future, depending on priorities and overall specific needs.
